src/renderers/WebGLRenderer.js

```diff
@@ -1514,11 +1508,11 @@ function WebGLRenderer( parameters = {} ) {
 		const environment = material.isMeshStandardMaterial ? scene.environment : null;
 		const encoding = ( _currentRenderTarget === null ) ? _this.outputEncoding : _currentRenderTarget.texture.encoding;
 		const envMap = ( material.isMeshStandardMaterial ? cubeuvmaps : cubemaps ).get( material.envMap || environment );
-		const vertexAlphas = material.vertexColors === true && !! object.geometry && !! object.geometry.attributes.color && object.geometry.attributes.color.itemSize === 4;
```

Comment:
I don't think the `!! object.geometry` make sense. Render items always have a geometry, no?

Reply:
Ah wait, we need this for `ImmediateRenderObject`.

Reply:
I wonder if we could remove `ImmediateRenderObject` and implemented the marching cube example more straightforward based in `BufferGeometry`. This would noticeably simplify the renderer. And in that way,`WebGPURenderer` does not have to support `ImmediateRenderObject`.

Reply:
That sounds good to me 👍